syntax = "proto3";


package helloworld;


// voting service

service VotingService {
  rpc RequestVote (VoteRequest) returns (VoteReply) {}
  rpc Heartbeat (HeartbeatRequest) returns (HeartbeatReply) {}
}

message VoteRequest {
  int32 term = 1;
  int32 candidateId = 2;
  int32 lastLogIndex = 3;
  int32 lastLogTerm = 4;
}

message VoteReply {
  int32 term = 1;
  bool voteGranted = 2;
}

message HeartbeatRequest {
  int32 term = 1;
  int32 leaderId = 2;
}

message HeartbeatReply {
  int32 term = 1;
  bool success = 2;
}